nano/lib/jsonconfig.hpp

```diff
@@ -163,6 +164,12 @@ class jsonconfig : public nano::error_aware<>
 		return child_config;
 	}
 
+	jsonconfig & operator= (jsonconfig const & other_a)
+	{
+		tree = other_a.tree;
```

Comment:
I don't think you need to define this as a default one will be provided for you by the compiler. Defining it also prevents move constructors being implicitly generated.

Reply:
I tried that but got "Object of type 'nano::jsonconfig' cannot be assigned because its copy assignment operator is implicitly deleted", probably by not following rule of 5. Also a defaulted impl yields the same error on clang. I think it's fine to keep this for now, I want to rewrite parts of this.

Reply:
Hmm weird, I did some more digging and actually the rules are more strict than I thought: http://eel.is/c++draft/class.copy.assign#7 looks like the tree& will stop them being generated.